app/js/lib/foxbox/network.js

```diff
@@ -187,94 +180,76 @@ export default class Network extends EventDispatcher {
+    // existing link's origin isn't changed.
+    if ((!origin && !link) || (link && link.origin === origin)) {
+      return;
     }
 
-    return this[p.ping](`${this.tunnelOrigin}/ping`)
-      .then((isOnline) => this[p.onPong](p.remote, isOnline));
-  }
+    // Let's destroy old link if origin is changed or not available anymore.
+    if (link) {
+      link.off('online', this[p.onLinkOnlineChange]);
+      link.disableAutoPing();
+      this[symbol] = null;
+    }
 
-  /**
-   * Pings both local and remote boxes simultaneously (if discovered).
-   *
-   * @private
-   */
-  [p.pingAllBoxes]() {
-    this[p.pingLocalBox]();
-    this[p.pingRemoteBox]();
+    if (origin) {
+      this[symbol] = new BoxLink(origin);
+      this[symbol].enableAutoPing(this[p.linkPingInterval]);
+      this[symbol].on('online', this[p.onLinkOnlineChange]);
+    }
   }
 
   /**
-   * Performs HTTP 'GET' request to the specified URL. Returns 'true' if
-   * response was successful (any of 200-299 status codes) or 'false'
-   * otherwise.
-   *
-   * @param {string} url The URL to send the request to.
-   * @return {Promise<boolean>}
+   * Updates box links whenever local or tunnel origin has changed.
+   * 
    * @private
    */
-  [p.ping](url) {
-    return fetch(url, { cache: 'no-store' })
-      .then((res) => res.ok)
-      .catch((error) => {
-        console.error('Error occurred while pinging content: %o', error);
-        return false;
-      });
+  [p.onLinkOriginChange]() {
+    this[p.updateLink](p.localLink, this.localOrigin);
+    this[p.updateLink](p.tunnelLink, this.tunnelOrigin);
   }
 
   /**
-   * Process ping response (pong). If 'online' state is changed we emit 'online'
-   * event. Since 'online' state depends on two factors: local and remote server
-   * availability, there is a slight chance that this method will cause two
-   * events e.g. if previously box was available only locally and now it's
-   * available only remotely we'll likely generate event indicating that box
-   * went offline following by another event indicating that box is online
-   * again.
-   *
-   * @param {Symbol} localOrRemote Symbol indicating whether we process local
-   * pong or remote one.
-   * @param {boolean} isOnline Flag that indicates whether pinged server
-   * successfully responded to ping request.
+   * Updates overall "online" status whenever either local or tunnel link
+   * "online" status has changed. If overall status has changed "online" event
+   * is emitted.
+   * 
    * @private
    */
-  [p.onPong](localOrRemote, isOnline) {
-    // If value hasn't changed, there is no reason to think that overall
-    // 'online' state has changed.
-    if (this[localOrRemote] === isOnline) {
-      return;
-    }
-
-    const previousOnlineState = this[p.local] || this[p.remote];
-
-    this[localOrRemote] = isOnline;
+  [p.onLinkOnlineChange]() {
+    const onlineStatus = !!this[p.localLink] && this[p.localLink].online ||
```

Comment:
The idea is to remove box link instance once we don't have origin for it anymore.